packages/lib/src/components/UPI/components/UPIComponent/UPIComponent.tsx

```diff
@@ -1,58 +1,104 @@
-import { h, Fragment, RefObject } from 'preact';
-import { useMemo, useRef, useState } from 'preact/hooks';
-import VpaInput from '../VpaInput';
-import ContentSeparator from '../../../internal/ContentSeparator';
-import Button from '../../../internal/Button';
+import { Fragment, h, RefObject } from 'preact';
+import { useCallback, useRef, useState } from 'preact/hooks';
 import useCoreContext from '../../../../core/Context/useCoreContext';
-import getImage from '../../../../utils/get-image';
 import { PayButtonFunctionProps, UIElementStatus } from '../../../types';
 import { VpaInputHandlers } from '../VpaInput/VpaInput';
+import VpaInput from '../VpaInput';
+import SegmentedControl from '../../../internal/SegmentedControl';
+import getImage from '../../../../utils/get-image';
+import { UpiMode } from '../../types';
+import './UPIComponent.scss';
 
 interface UPIComponentProps {
-    payButton: (props: PayButtonFunctionProps) => h.JSX.Element;
-    ref: (ref: RefObject<typeof UPIComponent>) => void;
-    onChange: ({ data: VpaInputDataState, valid, errors, isValid: boolean }) => void;
-    onSubmit(): void;
-    onGenerateQrCodeClick(): void;
+    defaultMode: UpiMode;
+    showPayButton: boolean;
+    ref(ref: RefObject<typeof UPIComponent>): void;
+    payButton(props: PayButtonFunctionProps): h.JSX.Element;
+    onChange({ data: VpaInputDataState, valid, errors, isValid: boolean }): void;
+    onUpdateMode(mode: UpiMode): void;
 }
 
-export default function UPIComponent({ onChange, onGenerateQrCodeClick, payButton }: UPIComponentProps): h.JSX.Element {
+const A11Y = {
+    ButtonId: {
+        VPA: `upi-button-${UpiMode.Vpa}`,
+        QR: `upi-button-${UpiMode.QrCode}`
+    },
+    AreaId: {
+        VPA: `upi-area-${UpiMode.Vpa}`,
+        QR: `upi-area-${UpiMode.QrCode}`
+    }
+};
+
+export default function UPIComponent({ defaultMode, onChange, onUpdateMode, payButton, showPayButton }: UPIComponentProps): h.JSX.Element {
     const { i18n, loadingContext } = useCoreContext();
     const inputRef = useRef<VpaInputHandlers>(null);
     const [status, setStatus] = useState<UIElementStatus>('ready');
-    const [isQrCodeFlow, setIsQrCodeFlow] = useState<boolean>(false);
+    const [mode, setMode] = useState<UpiMode>(defaultMode);
+    const isMobile = window.matchMedia('(max-width: 768px)').matches && /Android|iPhone|iPod/.test(navigator.userAgent);
```

Comment:
We should extract this function into utils. I'm sure it's going to be needed in some other place later.

Reply:
done!